daemon/alloc_hooks_jemalloc.cc

```diff
@@ -44,6 +44,38 @@ static int jemalloc_get_stats_prop(const char* property, size_t* value) {
     return je_mallctl(property, value, &size, NULL, 0);
 }
 
+// set je_malloc specific property
+static int je_set(const char* property, unsigned int value) {
+    return je_mallctl(property, NULL, NULL, (void*)&value, sizeof(value));
+}
+
+template<typename T>
+T je_get(const char* property) {
+    T value;
+    size_t size = sizeof(value);
+    value = 0;
+    int err = je_mallctl(property, (void*)&value, &size, NULL, 0);
+    if (err) {
+        get_stderr_logger()->log(EXTENSION_LOG_WARNING, NULL,
+                                 "je_get [%s] error %d - "
+                                 "could not determine narenas.",cb_strerror(err), err);
+    }
+    return value;
+}
+
+// je_malloc specific mib info
+template<typename T>
+T je_get_mib2(const char* propname, size_t i)  {
+    size_t mib[6];
+    T v;
+    size_t miblen = sizeof(mib) / sizeof(size_t);
+    size_t sz = sizeof(T);
+    je_mallctlnametomib(propname, mib, &miblen);
```

Comment:
Given the MIB lookup is supposed to be a bit slow, for production use (when associating threads  with arenas) we  want to cache the MIB lookup (nametomib) result and call je_mallctlbymib() directly.